### ISSUE TITLE
running multiple ThermInt algorithms in multiple threads causes DomainError

### ISSUE THREAD

**Issue by imLew** [theogf/ThermodynamicIntegration.jl#13]
Not sure if it makes sense to post the whole stack trace here. The end result is:
```
ERROR: TaskFailedException:
DomainError with -1.0:
log will only return a complex result if called with a complex argument. Try log(Complex(x)).
```

To reproduce start a julia REPL with `--threads 2` and run
```
using Distributions
using ThermodynamicIntegration

function do_therm_int()
    prior = MvNormal(ones(3))
    logprior(θ) = logpdf(prior, θ)
    loglikelihood(θ) = logpdf(prior, θ)
    alg = ThermInt(n_steps=30, n_samples=3000)
    logZ = alg(logprior, loglikelihood, rand(prior))
end

Threads.@threads for i in 1:5
    do_therm_int()
end
```

**Comment by theogf** [comment]
Hmm that sounds weird. I will try to reproduce it and see what's wrong. It's probably better anyway to parallelize over the different steps of the schedule.

**Comment by theogf** [comment]
Ok when I run it on 1.6, I get a different error, but it seems to come from `AdvancedHMC` I will try to fix it.

**Comment by theogf** [comment]
This should be solved by #15 I think
